collated/test/justintkj.md

```diff
@@ -0,0 +1,568 @@
+ * outside of the MapCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the MapCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+public class UndoCommandParserTest {
+
+    private UndoCommandParser parser = new UndoCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsUndoCommand() {
+        assertParseSuccess(parser, "1", new UndoCommand(1));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "a", String.format(MESSAGE_INVALID_COMMAND_FORMAT, UndoCommand.MESSAGE_USAGE));
+    }
+}
+```
+###### \java\systemtests\SortCommandSystemTest.java
+``` java
+public class SortCommandSystemTest extends AddressBookSystemTest {
+    @Test
+    public void sort() throws Exception {
+        Model model = getModel();
+
+        /* Case: Sort all persons by name, CamelCase */
+        String command = SortCommand.COMMAND_WORD + " nAmE";
+        String expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "name";
+        assertCommandSuccess(command, model, expectedResultMessage);
+
+        /* Case: Sort all persons by null*/
+        command = SortCommand.COMMAND_WORD + "  ";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_FAILURE + " \n" + SortCommand.MESSAGE_USAGE;
+        assertCommandFailure(command, expectedResultMessage);
+
+        /* Case: Sort all persons by name */
+        command = SortCommand.COMMAND_WORD + " name";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "name";
+        assertCommandSuccess(command, model, expectedResultMessage);
+
+        /* Case: Sort all persons by address */
+        command = SortCommand.COMMAND_WORD + " address";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "address";
+        model.sortPerson("address");
+        assertCommandSuccess(command, model, expectedResultMessage);
+
+        /* Case: Sort all persons by number */
+        command = SortCommand.COMMAND_WORD + " number";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "number";
+        model.sortPerson("number");
+        assertCommandSuccess(command, model, expectedResultMessage);
+
+        /* Case: Sort all persons by remark */
+        command = SortCommand.COMMAND_WORD + " remark";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "remark";
+        model.sortPerson("remark");
+        assertCommandSuccess(command, model, expectedResultMessage);
+
+        /* Case: Sort all persons by birthday */
+        command = SortCommand.COMMAND_WORD + " birthday";
+        expectedResultMessage = SortCommand.MESSAGE_SORT_SUCCESS + "birthday";
+        model.sortPerson("birthday");
+        assertCommandSuccess(command, model, expectedResultMessage);
```

Comment:
Try to minimize repetitive code blocks by extracting out similarities as utility methods.